polynomials/polyreal.11.cpp

```diff
@@ -194,6 +194,58 @@ struct poly_real {
             x *= scale;
         return rn;
     }
+
+    poly_real derive(){
+        vector<T> b;
+        for (int i = 1; i < size(); ++i){
+            b.push_back(a[i] * i);
+        }
+
+        if (b.empty())
+            return b = {0};
+
+        return poly_real(b);
+    }
+
+    poly_real integrate(){
+        vector<T> b = {0};
+        int n = size();
+        for (int i = 0; i < size(); ++i){
+            b.push_back(a[i]*1.0 / (i+1.0));
+        }
+
+        return {b};
+    }
+
+    void _trim(int n){
+        while (size() > n)
+            a.pop_back();
+    }
+
+    poly_real ex(){
```

Comment:
Please document the precondition that the length must be a power of two, and also that a[0] must be zero (otherwise the algorithm is incorrect) and change the name to poly_exp.